plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -0,0 +1,202 @@
+
+        Metadata metadata = new Metadata();
+        try {
+            byte[] decodedContent = Base64.decode(base64Input.getBytes(UTF_8));
+            String parsedContent = TikaImpl.parse(decodedContent, metadata, indexedChars);
+
+            if (fields.contains(Field.CONTENT)) {
+                // somehow tika seems to append a newline at the end automatically, lets remove that again
+                ingestDocument.setFieldValue(targetField + "." + Field.CONTENT.toLowerCase(), parsedContent.trim());
+            }
+
+            if (fields.contains(Field.LANGUAGE)) {
+                LanguageIdentifier identifier = new LanguageIdentifier(parsedContent);
+                String language = identifier.getLanguage();
+                ingestDocument.setFieldValue(targetField + "." + Field.LANGUAGE.toLowerCase(), language);
+            }
+
+            if (fields.contains(Field.DATE)) {
+                String createdDate = metadata.get(TikaCoreProperties.CREATED);
+                if (createdDate != null) {
+                    ingestDocument.setFieldValue(targetField + "." + Field.DATE.toLowerCase(), createdDate);
+                }
+            }
+
+            if (fields.contains(Field.TITLE)) {
+                String title = metadata.get(TikaCoreProperties.TITLE);
+                if (Strings.hasLength(title)) {
+                    ingestDocument.setFieldValue(targetField + "." + Field.TITLE.toLowerCase(), title);
+                }
+            }
+
+            if (fields.contains(Field.AUTHOR)) {
+                String author = metadata.get("Author");
+                if (Strings.hasLength(author)) {
+                    ingestDocument.setFieldValue(targetField + "." + Field.AUTHOR.toLowerCase(), author);
+                }
+            }
+
+            if (fields.contains(Field.KEYWORDS)) {
+                String keywords = metadata.get("Keywords");
+                if (Strings.hasLength(keywords)) {
+                    ingestDocument.setFieldValue(targetField + "." + Field.KEYWORDS.toLowerCase(), keywords);
+                }
+            }
+
+            if (fields.contains(Field.CONTENT_TYPE)) {
+                String contentType = metadata.get(Metadata.CONTENT_TYPE);
+                if (Strings.hasLength(contentType)) {
+                    ingestDocument.setFieldValue(targetField + "." + Field.CONTENT_TYPE.toLowerCase(), contentType);
+                }
+            }
+
+            if (fields.contains(Field.CONTENT_LENGTH)) {
+                String contentLength = metadata.get(Metadata.CONTENT_LENGTH);
+                String length = Strings.hasLength(contentLength) ? contentLength : String.valueOf(parsedContent.length());
+                ingestDocument.setFieldValue(targetField + "." + Field.CONTENT_LENGTH.toLowerCase(), length);
+            }
+        } catch (Throwable e) {
+            throw new ElasticsearchParseException("Error parsing document in field [{}]", e, sourceField);
```

Comment:
added more unit tests to check for empty documents, language gibberish (resulting in lt as before) and the french doc..